dataland-frontend/src/utils/FileUploadUtils.ts

```diff
@@ -0,0 +1,86 @@
+import { CompanyReport } from "@clients/backend";
+import { ApiClientProvider } from "@/services/ApiClients";
+import Keycloak from "keycloak-js";
+
+export interface DocumentToUpload {
+  file: File;
+  fileNameWithoutSuffix: string;
+
+  reference: string;
+}
+
+export interface StoredReport extends CompanyReport {
+  reportName: string;
+}
+
+export interface ReportToUpload extends CompanyReport {
+  file: File;
+  fileNameWithoutSuffix: string;
+}
+
+/**
+ * uploads Files through the frontend
+ * @param files the list of files to upload
+ * @param getKeycloakPromise getter for a keycloak promise
+ */
+export async function uploadFiles(
+  files: ReportToUpload[] | DocumentToUpload[],
+  getKeycloakPromise: () => Promise<Keycloak>
+): Promise<void> {
+  const documentControllerApi = await new ApiClientProvider(getKeycloakPromise()).getDocumentControllerApi();
+  for (const fileToUpload of files) {
+    const fileIsAlreadyInStorage = (await documentControllerApi.checkDocument(fileToUpload.reference)).data
```

Comment:
What happens if the document is already stored in the backend?

Reply:
Nothing. The user does not experience a difference. No further api call is done and the dataset gets a reference to the previously uploaded document (which has the same hash). However the name of the document stays like the user has uploaded it.

Reply:
ok